pkg/common/utils/ioutil_test.go

```diff
@@ -40,6 +40,22 @@ func TestIoutilCopyBuffer(t *testing.T) {
 	}
 }
 
+func TestIoutilCopyBufferWithNilBuffer(t *testing.T) {
+	var writeBuffer bytes.Buffer
+	src := bytes.NewBufferString("hertz is very good!!!")
+	dst := network.NewWriter(&writeBuffer)
+	// src.Len() will change, when use src.read(p []byte)
+	srcLen := int64(src.Len())
+	written, err := CopyBuffer(dst, src, nil)
+
+	if written != srcLen {
+		t.Fatalf("Unexpected written: %d. Expecting: %d", written, srcLen)
+	}
```

Comment:
Using assert.DeepEqual() would be a better choice. Below is the same.

Reply:
done

Reply:
> done

There's more to change.

Reply:
i am not sure if get what you mean. But if replaced all tests that use "if else" pattern in utils package with assert.DeepEqual.
Pls let me know if i made a mistake.